File: backend_dotnet/fruit_api/fruit_api/Models/OrderItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("OrderItems")]  // Sửa từ Order_items thành OrderItems
public class OrderItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("orderItemId")]  // Sửa từ order_item_id thành orderItemId
    public string OrderItemId { get; set; } = null!;

    [Required]
    [Column("orderId")]  // Sửa từ order_id thành orderId
    public string OrderId { get; set; } = null!;

    [Required]
    [Column("productId")]  // Sửa từ product_id thành productId
    public string ProductId { get; set; } = null!;

    [Required]
    [Column("quantity")]  // Giữ nguyên
    public int Quantity { get; set; }

    [Required]
    [Column("priceAtTime", TypeName = "decimal(12,2)")]  // Sửa từ price_at_time thành priceAtTime
    public decimal PriceAtTime { get; set; }

    [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
    [Column("subtotal")]  // Giữ nguyên
    public decimal Subtotal { get; private set; }

    // Navigation properties
    public Order? Order { get; set; }
    public Product? Product { get; set; }
}

File: frontend_flutter/fruit_market/lib/models/OrderItem.dart
import 'product.dart';
class OrderItem {
  final String orderItemId;
  final String orderId;
  final String productId;
  final String productName;  // THÊM
  final String? imageUrl;    // THÊM
  final int quantity;
  final double priceAtTime;
  final double subtotal;

  /// nếu API trả luôn product
  final Product? product;

  OrderItem({
    required this.orderItemId,
    required this.orderId,
    required this.productId,
    required this.productName,
    this.imageUrl,
    required this.quantity,
    required this.priceAtTime,
    required this.subtotal,
    this.product,
  });

  /// JSON -> Object
  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      orderItemId: json['orderItemId'] ?? '',
      orderId: json['orderId'] ?? '',
      productId: json['productId'] ?? '',
      productName: json['productName'] ?? '',
      imageUrl: json['imageUrl'],
      quantity: json['quantity'] ?? 0,
      priceAtTime: (json['price'] as num?)?.toDouble() ?? 0,
      subtotal: (json['subtotal'] as num?)?.toDouble() ?? 0,
      product: json['product'] != null ? Product.fromJson(json['product']) : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'orderItemId': orderItemId,
      'orderId': orderId,
      'productId': productId,
      'productName': productName,
      'imageUrl': imageUrl,
      'quantity': quantity,
      'priceAtTime': priceAtTime,
      'subtotal': subtotal,
      'product': product?.toJson(),
    };
  }
}